app/src/header/tabs.rs

```diff
@@ -19,11 +19,18 @@ impl<'a> Widget for Tabs<'a> {
 			tabs
 				.iter()
 				.enumerate()
-				.map(|(i, _)| {
+				.map(|(i, tab)| {
+					let mut text = format!(" {} ", i + 1);
+					if THEME.tab.max_width > 0 {
+						text
+							.push_str(&tab.current_name().chars().take(THEME.tab.max_width).collect::<String>());
```

Comment:
The number of chars and the display width are different, one Unicode char may take up 2 cells. it's better to use the `unicode-width` crate here.